### ISSUE TITLE
[BUG]: PPO training come across negative reward, could you please help me resolve this, appreciate.

### ISSUE THREAD

**Issue by Modas-Li** [hpcaitech/ColossalAI#3888]
### 🐛 Describe the bug

PPO training come across negative reward, could you please help me resolve this, appreciate.
My parameters set as below:

```bash
torchrun --standalone --nproc_per_node=1 train_prompts.py \
    --pretrain "/data/bloom-560m-xc-0523" --model 'bloom' \
    --strategy colossalai_zero2 \
    --prompt_dataset /data/huggingface_cache/datasets/data_xc_sample_ppo_0517.json \
    --pretrain_dataset /data/huggingface_cache/datasets/data_xc_0519.json \
    --rm_model 'bloom' \
    --rm_pretrain /data/ColossalAI/applications/Chat/to/bloom-560m \
    --rm_path /data/ColossalAI/applications/Chat/to/rmstatic-rm-20230602.pt \
    --train_batch_size 1 --experience_batch_size 1 \
    --max_epochs 5 --max_timesteps 100 --update_timesteps 10 \
    --num_episodes 20 --use_wandb \
    --save_path '/data/bloom-560m-xc-0523/ppo_0602.bin' --max_seq_len 512
```

![image](https://github.com/hpcaitech/ColossalAI/assets/40042370/212dd63a-e9b6-4f79-b876-91ea6e230da8)
